crates/matrix-sdk/src/notification_settings/rules.rs

```diff
@@ -210,11 +209,9 @@ impl Rules {
         }
     }
 
-    /// Insert a new `Room` push rule for a given `room_id` and return an
-    /// optional `Command` describing the action to be performed on the
-    /// user's account data.
-    pub(crate) fn insert_room_rule(
-        &mut self,
+    /// Build a command to insert a push rule
+    pub(crate) fn build_insert_rule_command(
```

Comment:
Ok, so first I thought all the `_command` methods don't really use `self` (directly or indirectly), and that suggests a slightly different code structuring here.

But when making a delete command, we shouldn't just let pass through an error where we're trying to remove a rule that doesn't exist. So it could be useful to keep a look at the current rules.

So we can go one of two ways:

- Either rename all these methods `prepare_{action}_command`, **but** have them do real validation.
- Or having a new `struct RuleCommands<'a> { commands: Vec<Command>, rules: &'a Ruleset }` that has many helper methods `insert_rule`, etc., and it's independent from the `Rules`? Then we change the signature of `Rules::apply` so it takes a `RuleCommands` parameter. Commands can be tested separately that way.